### PR TITLE
feat: [CDS-40429]: support Generic and Docker repository format for Azure Web App Artifactory Artifact

### PR DESCRIPTION
**Pull request by Tamaraius**
### Summary
support Generic and Docker repository format for Azure Web App Artifactory Artifact

#### Screenshots
https://user-images.githubusercontent.com/96610348/181005342-fbf05081-ee48-4982-aafc-379e272919a0.mov

#### PR Checklist

<details>
<summary>Please check if your PR fulfils the following requirements</summary>

- [ ] Tests for the changes have been added. Ideally, include a test that fails without this PR but passes with it.
- [ ] Docs have been [added/updated](https://harness.atlassian.net/jira/software/c/projects/DOC/boards/40).
</details>

<details>
<summary>Use the following comments to re-trigger PR Checks</summary>

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- Standards: `retrigger standards`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- Coverage: `retrigger coverage`
- Rebase: `trigger rebase`
- Cypress: `trigger cypress`
- Fix Prettier: `fix prettier`
</details>

### CODE REVIEW

**Comment by Tamaraius** [comment]
retrigger coverage

File: src/modules/75-cd/components/PipelineSteps/K8sServiceSpec/ArtifactSource/ArtifactoryArtifactSource/ArtifactoryArtifactSource.tsx [hunk 1]
```diff
@@ -92,15 +100,17 @@ const TagFields = (props: TagFieldsProps): JSX.Element => {
     selectedDeploymentType === 'WinRm' ||
     selectedDeploymentType === 'Ssh'
 
+  const isGenericArtifactory = isServerlessOrSshOrWinRmSelected || isAzureWebAppGenericSelected
```
Comment (chetanvekariya-harness) [inline src/modules/75-cd/components/PipelineSteps/K8sServiceSpec/ArtifactSource/ArtifactoryArtifactSource/ArtifactoryArtifactSource.tsx:103]: Pass this prop to the component. This logic is being duplicated in both the components

File: src/modules/75-cd/components/PipelineSteps/K8sServiceSpec/ArtifactSource/ArtifactoryArtifactSource/__tests__/ArtifactoryArtifactSource.test.tsx [hunk 2]
```diff
@@ -108,7 +111,6 @@ describe('Artifactory Artifact Source tests', () => {
     expect(await waitFor(() => findByText('pipeline.artifactPathLabel'))).toBeInTheDocument()
     expect(await waitFor(() => artifactSourceUtils.fromPipelineInputTriggerTab)).toBeCalled()
     expect(await waitFor(() => artifactSourceUtils.isFieldfromTriggerTabDisabled)).toBeCalled()
```
Comment (chetanvekariya-harness) [inline src/modules/75-cd/components/PipelineSteps/K8sServiceSpec/ArtifactSource/ArtifactoryArtifactSource/__tests__/ArtifactoryArtifactSource.test.tsx:113]: Why is this removed ?
`expect(await waitFor(() => cdng.useGetBuildDetailsForArtifactoryArtifactWithYaml)).toBeCalled()`
Existing functionality should not change after your code change, I guess.

**Review by chetanvekariya-harness — APPROVED** [review]

**Review by bhavya-sinha — APPROVED** [review]

File: src/modules/70-pipeline/components/ArtifactsSelection/ArtifactRepository/ArtifactLastSteps/Artifactory/__tests__/Artifactory.test.tsx [hunk 3]
```diff
@@ -307,3 +314,44 @@ describe('Serverless artifact', () => {
...
+    expect(repositoryFormat!).toBeDefined()
+    expect(repositoryFormat!).toHaveAttribute('value', 'Generic')
+
+    const artifactDirectory = container.querySelector('input[name="artifactDirectory"]')
+    expect(artifactDirectory!).toBeDefined()
+
+    const repositoryUrl = container.querySelector('input[name="repositoryUrl"]')
+    expect(repositoryUrl!).toBeNull()
+
+    expect(container).toMatchSnapshot()
```
[the remark this answers is not in the document]
Comment (Tamaraius) [reply src/modules/70-pipeline/components/ArtifactsSelection/ArtifactRepository/ArtifactLastSteps/Artifactory/__tests__/Artifactory.test.tsx:336]: it makes sense, will take care of it and future use cases in the upcoming PRs, thanks :)

[the remark this answers is not in the document]
Comment (Tamaraius) [reply src/modules/70-pipeline/components/ArtifactsSelection/ArtifactRepository/ArtifactLastSteps/Artifactory/__tests__/Artifactory.test.tsx]: removed :)

File: src/modules/72-triggers/pages/triggers/utils/TriggersWizardPageUtils.ts [hunk 4]
```diff
@@ -1577,7 +1599,11 @@ export const getArtifactTableDataFromData = ({
     // To decide whether stage is serverless or not serverless below function is called
     // If stage is serverless, there is not such field as "tag" instead simialr field name called "artifactPath" is present
     // Similarly, there is not such field as "tagRefex" instead simialr field name called "artifactPathFilter" is present
+    // Same applies to generic azure web app
     const isServerlessDeploymentTypeSelected = isServerlessDeploymentStage(stageId, pipeline)
+    const isAzureWebAppGenericTypeSelected = isAzureWebAppGenericDeploymentStage(stageId, pipeline)
+
+    const isServerlessOrAzureWebAppGeneric = isServerlessDeploymentTypeSelected || isAzureWebAppGenericTypeSelected
```
[the remark this answers is not in the document]
Comment (Tamaraius) [reply src/modules/72-triggers/pages/triggers/utils/TriggersWizardPageUtils.ts]: removed :)

File: src/modules/75-cd/components/PipelineSteps/K8sServiceSpec/ArtifactSource/ArtifactoryArtifactSource/ArtifactoryArtifactSource.tsx (hunk 1 above)
Comment (Tamaraius) [reply src/modules/75-cd/components/PipelineSteps/K8sServiceSpec/ArtifactSource/ArtifactoryArtifactSource/ArtifactoryArtifactSource.tsx]: passed :)

**Review by bhavya-sinha — APPROVED** [review]

**Comment by Tamaraius** [comment]
retrigger titlecheck

File: src/modules/72-triggers/pages/triggers/utils/TriggersWizardPageUtils.ts [hunk 5]
```diff
@@ -1741,7 +1741,7 @@ export const getArtifactTableDataFromData = ({
             getString,
             isStageOverrideManifest: false,
             isManifest,
-            isServerlessDeploymentTypeSelected: isServerlessOrAzureWebAppGeneric
+            isServerlessDeploymentTypeSelected: isGenericArtifactory
```
Comment (chetanvekariya-harness) [inline src/modules/72-triggers/pages/triggers/utils/TriggersWizardPageUtils.ts:1744]: isGenericArtifactory: isGenericArtifactory

**Review by chetanvekariya-harness — APPROVED** [review]

**Review by kapilkumawat86 — APPROVED** [review]
